src/main/java/io/vertx/core/http/impl/ClientConnection.java

```diff
@@ -420,4 +559,19 @@ public void channelRead(ChannelHandlerContext chctx, Object msg) throws Exceptio
     });
     return socket;
   }
+
+  @Override
+  public HttpClientConnection connection() {
+    return this;
+  }
+
+  @Override
+  public HttpVersion version() {
+    return null;
```

Comment:
good catch, thanks and I will add tests for these.



> On Apr 6, 2016, at 10:42 AM, Clement Escoffier <notifications@github.com> wrote:
> 
> In src/main/java/io/vertx/core/http/impl/ClientConnection.java <https://github.com/eclipse/vert.x/pull/1353#discussion_r58670457>:
> 
> > @@ -420,4 +559,19 @@ public void channelRead(ChannelHandlerContext chctx, Object msg) throws Exceptio
> >      });
> >      return socket;
> >    }
> > +
> > +  @Override
> > +  public HttpClientConnection connection() {
> > +    return this;
> > +  }
> > +
> > +  @Override
> > +  public HttpVersion version() {
> > +    return null;
> not vertxVersion ?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub <https://github.com/eclipse/vert.x/pull/1353/files/bf875485313cf1f6ae199e2ab525539ed5676a2e#r58670457>